source/operators/frame_extractor/frame_extractor.py

```diff
@@ -0,0 +1,162 @@
+        s3.download_fileobj(s3bucket, s3key, f)
+    f.close()
+    return temp_chunk
+
+def put_record_s3(s3bucket, asset_id, chunk_id, frames):
+    file_name = 'private/assets/{}/output/{}/frame_list.json'.format(asset_id, chunk_id)
+    try:
+        s3.put_object( 
+            ACL='bucket-owner-full-control', 
+            Bucket=s3bucket, 
+            Key=file_name, 
+            Body=json.dumps({"chunk": frames}).encode()
+        )
+    except Exception as e:
+        print("put_record_s3: Failed saving file %s" % file_name)
+        print (e)
+
+def lambda_handler(event, context):
+    print("We got the following event:\n", event)
+    output_object = MediaInsightsOperationHelper(event)
+
+    if store_frames not in ["all", "original", "resized"]:
+        return "Invalid STORE_FRAMES option: %s (Valid: all, original, resized)" % store_frames
+
+    store_original_frames = store_frames in ["all", "original"]
+    store_resized_frames = store_frames in ["all", "resized" ]
+
+    try:
+        if "Video" in event["Input"]["Media"]:
+            s3bucket = event["Input"]["Media"]["Video"]["S3Bucket"]
+            s3key = event["Input"]["Media"]["Video"]["S3Key"]
+        workflow_id = str(event["WorkflowExecutionId"])
+        asset_id = event['AssetId']
+    except Exception:
+        output_object.update_workflow_status("Error")
+        output_object.add_workflow_metadata(FrameExtractionError="No valid inputs")
+        raise MasExecutionError(output_object.return_output_object())
+    print("Processing s3://"+s3bucket+"/"+s3key)
+    valid_video_types = [".avi", ".mp4", ".mov", ".ts"]
+    file_type = os.path.splitext(s3key)[1]
+
+    print("Event:\n", event)
+    
+    if file_type in valid_video_types:
+        # Extract frames and store paths in a list to persist on the dataplane
+        file_name = download_chunk(s3bucket, s3key)
+
+        cap = cv2.VideoCapture(file_name)
+        
+        metadata = {
+            'original_frame_width': cap.get(cv2.CAP_PROP_FRAME_WIDTH),
+            'original_frame_height': cap.get(cv2.CAP_PROP_FRAME_HEIGHT),
+            'frame_width': frame_resize_width,
+            'frame_height': frame_resize_height,
+            'fourcc': cap.get(cv2.CAP_PROP_FOURCC),
+            'frame_count': int(cap.get(cv2.CAP_PROP_FRAME_COUNT)),
+            'format': cap.get(cv2.CAP_PROP_FORMAT),
+            'mode': cap.get(cv2.CAP_PROP_MODE),
+            'fps': cap.get(cv2.CAP_PROP_FPS),
+            'workflow_id': workflow_id,
+            'asset_id': asset_id,
+            'store_frames': store_frames
+        }
+        
+        # Lists to capture timestamps
+        timestamp = []
+        store_timestamps = {}
+        
+        original_frame_keys = []
+        resized_frame_keys = []
+        frame_counter = 0
+        chunk_id = os.path.basename(s3key).split('.')[0]
+        while(cap.isOpened()):
+            _, frame = cap.read()
+
+            if _ == False:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `==` and `!=` operators use the compared objects&#39; ` __eq__` method to test if they are equal. To check if an object is a singleton,such as `None`, we recommend that you use the `is` identity comparison operator.

[Learn more](https://docs.python.org/3/reference/expressions.html#is)